plugins/setoid_ring/g_newring.mlg

```diff
@@ -89,7 +89,9 @@ VERNAC COMMAND EXTEND AddSetoidRing CLASSIFIED AS SIDEFF
   | [ "Print" "Rings" ] => { Vernacextend.classify_as_query } -> {
     Feedback.msg_notice (strbrk "The following ring structures have been declared:");
     Spmap.iter (fun fn fi ->
-      let sigma, env = Pfedit.get_current_context () in
+      (* We use the global env here as this shouldn't contain proof
+         data. *)
+      let sigma, env = let e = Global.env () in Evd.from_env e, e in
```

Comment:
wrong env

Reply:
We can access the proof state (if it exists) and take the right env here, can't we?

Reply:
@ejgallego ping, I think an `Option.cata` on the proof state, preserving previous behavior, would be better.

Reply:
This is still to be fixed.